### PR TITLE
Improve your performance with maps

### PR DESCRIPTION
**Pull request by chdeps**
@Minishlink : j'ai rajouté cela en tant que sous partie de ton dev standards sur les perf react native. Dis moi si tu penses que l'on devrait le mettre ailleurs ;)

### CODE REVIEW

**Review by Minishlink — APPROVED** [review]
C'est top ce résumé ! Je pense que tu peux en faire un fichier à part et linker dans l'article ;) Tu peux modifier sur gitbook d'ailleurs !

File: performance/front/react-native-performance.s.md [hunk 1]
```diff
@@ -57,6 +57,74 @@ If your app takes too much time to initialize, solve the problem incrementally:
...
+            && isRegionWithIn(boundaries(region), this.state.queriedRegion[key])) return;
+        //Call for new pins with debounce if the pinTypes or search parameters have changed since last time
+        //Or if the region queried is larger than the previous one
+        return this._callForPinsDebounce(region, pinTypes, search);
+      };
+```
+
+> As regards to caching, there still things that need to be done. At the time, I was doing that Apollo did not provide any caching per parameter. It would allow us to have a greater history of calls in cache & only the last one. Also, ideally you could think of caching when zooming out. This involves some dev on the back-end, as it means that you are capable of fetching pins in a ring & only inside a cercle.
+
+4. `componentShouldUpdate` is your friend. Don't re-render the map if your pins haven't updated for instance.
```
Comment (thibautguedou3) [inline performance/front/react-native-performance.s.md]: shouldComponentUpdate pour les puristes :)

File: performance/front/react-native-performance.s.md [hunk 2]
```diff
@@ -57,6 +57,74 @@ If your app takes too much time to initialize, solve the problem incrementally:
...
+5. Clustering of pins on the front end ([Clustering with react-native-maps](https://github.com/bamlab/react-native-components-collection/tree/master/packages/react-native-component-map-clustering))
```
Comment (thibautguedou3) [inline performance/front/react-native-performance.s.md]: Tu voudrais pas mettre un truc un peu général en plus ou pour compléter ce point en disant qu'il ne faut pas afficher trop d'infos sur une map?

**Comment by bam-bot-tech** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/feat/perfOnMaps/performance/front/react-native-maps-performance.s.md">react-native-maps-performance.s.md</a>**: Standard doesn't have a *Checks* part, how could you call this an Standard?
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/feat/perfOnMaps/performance/front/react-native-maps-performance.s.md">react-native-maps-performance.s.md</a>**: You failed to mention a *Bad Examples*
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/feat/perfOnMaps/performance/front/react-native-maps-performance.s.md">react-native-maps-performance.s.md</a>**: You failed to mention a *Good Examples* 
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/feat/perfOnMaps/performance/front/react-native-maps-performance.s.md">react-native-maps-performance.s.md</a>**: Does not seem to be included in the root readme
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/feat/perfOnMaps/performance/front/react-native-maps-performance.s.md">react-native-maps-performance.s.md</a>**: Does not seem to be included in the root summary
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**